### PR TITLE
added Custom RankCard System

### PR DESCRIPTION
**Pull request by cyklon73**
## Checklist

 - [X] Read [Contribution Guidelines](https://github.com/SlimeCloud/java-SlimeBot#-contributing) and [Style Guides](https://github.com/SlimeCloud/java-SlimeBot#-style-guide)
 - [ ] Test the Code
 - [x] This PR is ready to review and merge

## Description
Describe what you have changed

## Related Issue
N/A

## Other Information
Custom RankCard System implemented.
Commands still missing

### CODE REVIEW

**Review by MatitoDA — CHANGES_REQUESTED** [review]
[siehe dc](https://discord.com/channels/1077255218728796192/1099133413736317100/1145745715088326806)

File: README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ statt dem Initialisieren der Funktion eine Warnung ausgeben.
 ## 🗄️ Datenbank
 Wir verwenden eine [PostgreSQL](https://www.postgresql.org/) Datenbank, um große Datenmengen zu speichern.
 Zur Interaktion mit der Datenbank verwenden wir [JDBI-Bibliothek](https://jdbi.org/).<br>
-Um mit der Datenbank zu interagieren wird eine sogenannte `DataClass` benötigt. Diese Klasse enthält alle daten die gespeichert werden sollen als Variabeln (es werden nur primitive typen und Strings unterstützt).
+Um mit der Datenbank zu interagieren wird eine sogenannte `DataClass` benötigt. Diese Klasse enthält alle daten die gespeichert werden sollen als Variabeln (es werden nur primitive typen, Strings und Enums unterstützt).
```
Comment (MineKing9534) [inline README.md:122]: daten -> Daten,
typen -> Typen (Ich seh grad, dass war vorher auch schon falsch, aber fix das bitte trotzdem xD)

File: src/main/java/com/slimebot/level/RankCard.java [hunk 2]
```diff
@@ -11,13 +16,19 @@
 import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.net.URL;
+import java.util.Map;
+
+import static com.slimebot.level.profile.Style.ROUND;
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/level/RankCard.java:21]: Please don't use a static import here, just use Style.ROUND when the enum constant is used

**Review by MineKing9534 — CHANGES_REQUESTED** [review]

**Review by MineKing9534 — CHANGES_REQUESTED** [review]
Please do a IntelliJ reformat code

**Review by MineKing9534 — APPROVED** [review]

**Review by MatitoDA — CHANGES_REQUESTED** [review]
If you Change the Background, there a some bugs:

- if you set an Image as Background and then want to set a Color as background, the Image does not disappear.
- if you set an invalid image link/color, there is no feedback for the user

File: src/main/java/com/slimebot/commands/level/card/CardCommand.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+
+import java.awt.*;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@ApplicationCommand(name = "card", description = "passe deine rank card an", feature = "level")
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/level/card/CardCommand.java:23]: Passe deine Rankcard an

File: src/main/java/com/slimebot/commands/level/card/CardCommand.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+public class CardCommand {
+
+	private static CardProfile loadProfile(Member member) {
+		Supplier<CardProfile> sup = () -> new CardProfile(member.getGuild().getIdLong(), member.getIdLong());
+		return DataClass.load(sup, Map.of("guild", member.getGuild().getIdLong(), "user", member.getIdLong())).orElseGet(sup);
+	}
+
+
+	@ApplicationCommand(name = "edit", description = "bearbeite deine rankcard")
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/level/card/CardCommand.java:32]: Bearbeite deine Rankcard

File: src/main/java/com/slimebot/commands/level/card/CardCommand.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+					.addFrame("background.modal", BackgroundModalFrame::new)
+					.addFrame("progressbar", ProgressbarFrame::new)
+					.addFrame("progressbar.color", ProgressbarColorFrame::new)
+					.addFrame("border", BorderFrame::new)
+					.addFrame("reset", ResetWarningFrame::new).start(new CallbackState(event), "main");
+		}
+
+	}
+
+	@ApplicationCommand(name = "info", description = "zeigt deine aktuellen rankcard optionen an")
```
Comment (MineKing9534) [inline src/main/java/com/slimebot/commands/level/card/CardCommand.java:51]: Zeigt deine aktuellen Rankcard Optionen an

**Review by MineKing9534 — CHANGES_REQUESTED** [review]

**Review by MatitoDA — APPROVED** [review]

**Review by MineKing9534 — CHANGES_REQUESTED** [review]
Please do a reformat with tabs enabled, you replace all indentations with spaces

**Review by MineKing9534 — APPROVED** [review]

**Pull request merged**